core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/HostEndpointUtils.scala

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+import com.microsoft.azure.synapse.ml.logging.common.CommonUtils
+import com.microsoft.azure.synapse.ml.logging.Usage.FabricConstants.{Capacities, WorkloadEndpointAutomatic, Workloads}
+import com.microsoft.azure.synapse.ml.logging.Usage.FabricConstants.{WebApi, WorkloadEndpointMl, WorkspaceID}
+import com.microsoft.azure.synapse.ml.logging.common.WebUtils.usageGet
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+import spray.json.{JsValue, JsonParser}
+
+object HostEndpointUtils {
```

Comment:
you have alot of objects everywhere. Can any of these be traits so you can compose them?